### PR TITLE
Add DNS cache

### PR DESCRIPTION
**Pull request by ditas**
Add DNS caching feature into the lhttpc client. DNS caching can be configured via sys.config -> lhttpc

### CODE REVIEW

File: src/lhttpc_sup.erl [hunk 1]
```diff
@@ -26,7 +26,7 @@
 %%% ----------------------------------------------------------------------------
 
 %%------------------------------------------------------------------------------
-%%% @author Oscar Hellstr�m <oscar@hellstrom.st>
+%%% @author Oscar Hellstr�m <oscar@hellstrom.st>
```
Comment (lrascao) [inline src/lhttpc_sup.erl:29]: your editor probably screwed this up

File: src/lhttpc_sock.erl [hunk 2]
```diff
@@ -27,7 +27,7 @@
 
 %%------------------------------------------------------------------------------
 %%% @private
-%%% @author Oscar Hellstr�m <oscar@hellstrom.st>
+%%% @author Oscar Hellstr�m <oscar@hellstrom.st>
```
Comment (lrascao) [inline src/lhttpc_sock.erl:30]: your editor probably screwed this up

File: src/lhttpc_client.erl [hunk 3]
```diff
@@ -27,7 +27,7 @@
 
 %%------------------------------------------------------------------------------
 %%% @private
-%%% @author Oscar Hellstr�m <oscar@hellstrom.st>
+%%% @author Oscar Hellstr�m <oscar@hellstrom.st>
```
Comment (lrascao) [inline src/lhttpc_client.erl:30]: your editor probably screwed this up

File: src/lhttpc.erl [hunk 4]
```diff
@@ -26,7 +26,7 @@
 %%% ----------------------------------------------------------------------------
 
 %%------------------------------------------------------------------------------
-%%% @author Oscar Hellstr�m <oscar@hellstrom.st>
+%%% @author Oscar Hellstr�m <oscar@hellstrom.st>
```
Comment (lrascao) [inline src/lhttpc.erl:29]: your editor probably screwed this up

File: src/lhttpc.erl [hunk 5]
```diff
@@ -437,6 +437,11 @@ request(URL, Method, Hdrs, Body, Timeout, Options) ->
 -spec request(string(), port_num(), boolean(), string(), method(),
     headers(), iodata(), pos_timeout(), options()) -> result().
 request(Host, Port, Ssl, Path, Method, Hdrs, Body, Timeout, Options) ->
+    
+%%    io:format("------------REQUEST HOST0 ~p ~p~n", [Host0, ?MODULE]),
+%%    Host = simple_mem_watcher:get_cached_dns(Host0),
+%%    io:format("------------REQUEST HOST ~p ~p~n", [Host, ?MODULE]),
```
Comment (lrascao) [inline src/lhttpc.erl:443]: this needs to be dropped

File: src/lhttpc_sock.erl [hunk 6]
```diff
@@ -65,9 +65,11 @@
 %%------------------------------------------------------------------------------
 -spec connect(host(), integer(), socket_options(), timeout(), boolean()) ->
     {ok, socket()} | {error, atom()}.
-connect(Host, Port, Options, Timeout, true) ->
+connect(Host0, Port, Options, Timeout, true) ->
+    Host = dns_cache_handler:get_cached_dns(Host0),
```
Comment (lrascao) [inline src/lhttpc_sock.erl:69]: is this a required feature? meaning i'm forced to use it, could it be made optional via the `Options` var?

File: rebar.config [hunk 7]
```diff
@@ -35,6 +35,10 @@
   deprecated_functions
  ]}.
 
+{relx,
+  [{sys_config, "config/sys.config"}
```
Comment (lrascao) [inline rebar.config:39]: this has been a lib up until now and not a release, why does that need to change?

File: config/sys.config [hunk 8]
```diff
@@ -0,0 +1,17 @@
+[
```
Comment (lrascao) [inline config/sys.config:1]: this has been a lib up until now and not a release, why does that need to change?

**Review by lrascao — CHANGES_REQUESTED** [review]

**Comment by lrascao** [comment]
also, commit history needs cleaning up

File: src/lhttpc_sock.erl (hunk 6 above)
Comment (ditas) [reply src/lhttpc_sock.erl:69]: It's not required. Actually, nothing happens if current domain is not in the list of domains to be cached ({domains_to_cache, ["dynamodb.us-west-2.amazonaws.com"]}). So if this feature is already optional through this domains_to_cache list

**Pull request closed without merging**